File: code/cs/OpcValue.cs
// DOC
OpcValue value = client.ReadNode("ns=2;s=Machine/Job/Speed");

// DOC
if (value.Status.IsGood) {
    // Your code to operate on the value.
}

// DOC
int intValue = (int)value.Value;

// DOC
int[] intValues = (int[])value.Value;

// DOC
OpcStatus status = client.WriteNode("ns=2;s=Machine/Job/Speed", 1200);

// DOC
int[] values = new int[3] { 1200, 1350, 1780 };
OpcStatus status = client.WriteNode("ns=2;s=Machine/Job/Speeds", values);

// DOC
if (!status.IsGood) {
    // Your code to handle a failed write operation.
}

// DOC
using (var client = new OpcClient("opc.tcp://localhost:4840")) {
    client.Connect();
    OpcValue arrayValue = client.ReadNode("ns=2;s=Machine/Job/Speeds");
 
    if (arrayValue.Status.IsGood) {
        int[] intArrayValue = (int[])arrayValue.Value;
 
        intArrayValue[2] = 100;
        intArrayValue[4] = 200;
        intArrayValue[9] = 300;
 
        OpcStatus status = client.WriteNode("ns=2;s=Machine/Job/Speeds", intArrayValue);
 
        if (!status.IsGood)
            Console.WriteLine("Failed to write array value!");
    }
}

File: code/vb/OpcValue.vb
'DOC
Dim value As OpcValue = client.ReadNode("ns=2;s=Machine/Job/Speed")

'DOC
If value.Status.IsGood Then
    'Your code to operate on the value.
End If

'DOC
Dim intValue As Integer = CInt(value.Value)

'DOC
Dim intValues As Integer() = CType(value.Value, Integer())

'DOC
Dim status As OpcStatus = client.WriteNode("ns=2;s=Machine/Job/Speed", 1200)

'DOC
Dim values As Integer() = New Integer(2) {1200, 1350, 1780}
Dim status As OpcStatus = client.WriteNode("ns=2;s=Machine/Job/Speeds", values)

'DOC
If Not status.IsGood Then
    'Your code to handle a failed write operation.
End If

'DOC
Using client As New OpcClient("opc.tcp://localhost:4840")
    client.Connect()
    Dim arrayValue As OpcValue = client.ReadNode("ns=2;s=Machine/Job/Speeds")

    If arrayValue.Status.IsGood Then
        Dim intArrayValue As Integer() = CType(arrayValue.Value, Integer())

        intArrayValue(2) = 100
        intArrayValue(4) = 200
        intArrayValue(9) = 300

        Dim status As OpcStatus = client.WriteNode("ns=2;s=Machine/Job/Speeds", intArrayValue)

        If Not status.IsGood Then _
            Console.WriteLine("Failed to write array value!")
    End If
End Using
